datasets/cfq/data_generation.py

```diff
@@ -0,0 +1,341 @@
+            if not short_output:
+                final_output = cur_output + " " + END_TOKEN
+            else:
+                final_output = partial_output + " " + END_TOKEN
+            if i == len(list_of_outputs) - 1:
+                final_output = final_output.replace(END_TOKEN, END_ITER_TOKEN)
+                final_output += " " + END_TOKEN
+        
+        if "test" not in split:        
+            output_file.write(START_TOKEN + " " + final_output + "\n")
+        
+        cur_input = question + " " + IN_OUT_TOKEN 
+        if short_input:
+            cur_input += " " + partial_output + " " + END_TOKEN
+        else:
+            cur_input += " " + cur_output + " " + END_TOKEN
+      
+      if "test" in split:
+        final_output = START_TOKEN + " " + cur_output 
+        final_output += " " + END_ITER_TOKEN + " " + END_TOKEN 
+        output_file.write(final_output + "\n")
+    
+    input_file.close()
+    output_file.close()
+    if filename_ops is not None:
+        ops_file.close()
+
+
+def main():        
+    train_ds = tfds.load("cfq/mcd1", split="train")
+    train_filename_in = "train.src"
+    train_filename_out = "train.tgt"
+    generate_seq2seq_examples(train_ds, train_filename_in, train_filename_out, 
+                             simplify_output=False, permute=True, 
+                             respect_question_order=True, random=True)
+    it_dec_train_filename_in = "it_dec_train.src"
+    it_dec_train_filename_out = "it_dec_train.tgt"
+    generate_it_dec_examples(train_ds, "train", it_dec_train_filename_in, 
+                             it_dec_train_filename_out, simplify_output=False, 
+                             permute=True, respect_question_order=True, random=True)
+    
+    val_ds = tfds.load("cfq/mcd1", split="test[:10%]")
+    val_filename_in = "val.src"
+    val_filename_out = "val.tgt"
+    generate_seq2seq_examples(val_ds, val_filename_in, val_filename_out,
+                              simplify_output=False, random=True)
+    it_dec_val_filename_in = "it_dec_val.src"
+    it_dec_val_filename_out = "it_dec_val.tgt"
+    generate_it_dec_examples(val_ds, "val", it_dec_val_filename_in, it_dec_val_filename_out,
+                             simplify_output=False, random=True)
+    
+    test_ds = tfds.load("cfq/mcd1", split="test")
+    test_filename_in = "test.src"
+    test_filename_out = "test.tgt"
+    generate_seq2seq_examples(test_ds, test_filename_in, test_filename_out, 
+                              simplify_output=False, random=True)
+    it_dec_test_filename_in = "it_dec_test.src"
+    it_dec_test_filename_out = "it_dec_test.tgt"
+    it_dec_test_filename_ops = "it_dec_test.ops"
+    generate_it_dec_examples(test_ds, "test", it_dec_test_filename_in, it_dec_test_filename_out, 
+                             it_dec_test_filename_ops, simplify_output=False, random=True)
+                           
+                             
+if __name__ == '__main__':
```

Comment:
Nit: since you use double quotes in this file
```suggestion
if __name__ == "__main__":
```